src/ChangeSet.php

```diff
@@ -216,6 +216,22 @@ public function removeObject(DataObject $object)
             // TODO: Handle case of implicit added item being removed.
 
             $item->delete();
+
+            // Get the implicitly included items for this ChangeSet
+            $implicit = $this->calculateImplicit();
+
+            foreach ($this->Changes()->filter(['Added' => ChangeSetItem::IMPLICITLY]) as $changeSetItem) {
+                $objectKey = $this->implicitKey($changeSetItem);
+
+                // If a ChangeSetItem exists, but isn't in $implicit, it's no longer required, so delete it
+                if (!array_key_exists($objectKey, $implicit ?? [])) {
```

Comment:
Null coalescing fallback not needed here as `calculateImplicit` should return an array already. If not, it should be added there rather than here.

```suggestion
                if (!array_key_exists($objectKey, $implicit)) {
```

Reply:
Updated